qiskit/providers/aer/primitives/estimator.py

```diff
@@ -0,0 +1,151 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Estimator class.
+"""
+
+from __future__ import annotations
+
+from collections.abc import Iterable, Sequence
+
+import numpy as np
+from qiskit.circuit import Parameter, QuantumCircuit
+from qiskit.compiler import transpile
+from qiskit.exceptions import QiskitError
+from qiskit.opflow import PauliSumOp
+from qiskit.primitives import EstimatorResult
+from qiskit.primitives.utils import init_circuit, init_observable
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from .. import AerSimulator
+from .base_estimator import BaseEstimator  # TODO: fix import path after Terra 0.21.
+
+
+class Estimator(BaseEstimator):
+    """
+    Aer implmentation of Estimator.
+
+    :Run Options:
+        - **shots** (None or int) --
+          The number of shots. If None, it calculates the exact expectation
+          values. Otherwise, it samples from normal distributions with standard errors as standard
+          deviations using normal distribution approximation.
```

Comment:
I'm wondering what is the correct behavior when `noise_model` is set to `AerSimulator`. If `densitymatrix` method is not used, estimator should take shots (with the default number `1024`?)  and calculate expval with observables (which will produce additional noise).

Reply:
After offline discussion with @hhorii,  I'll make other options (that does not use normal distribution approximation).

Reply:
a8c4231